doc_source/update-cluster.md

```diff
@@ -300,3 +303,151 @@ The previous command may use different default values from what is set in your c
+
+`kube-proxy --resource-containers` flag has been deprecated in [Kubernetes 1.16](https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG/CHANGELOG-1.16.md#deprecations-and-removals). If your cluster was created and upgraded since EKS 1.11, your `kube-proxy` daemonset spec has `--resource-containers` flag. This fails in Kubernetes 1.16, and must be removed before upgrading to EKS 1.16. We highly recommend creating `"kube-proxy-config"` configmap and update `kube-proxy` daemonset spec as follows:
+
+```
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: kube-proxy-config
+  namespace: kube-system
+  labels:
+    k8s-app: kube-proxy
+    eks.amazonaws.com/component: kube-proxy
+data:
+  config: |-
+    apiVersion: kubeproxy.config.k8s.io/v1alpha1
+    bindAddress: 0.0.0.0
+    clientConnection:
+      acceptContentTypes: ""
+      burst: 10
+      contentType: application/vnd.kubernetes.protobuf
+      kubeconfig: /var/lib/kube-proxy/kubeconfig
+      qps: 5
+    clusterCIDR: ""
+    configSyncPeriod: 15m0s
+    conntrack:
+      max: 0
+      maxPerCore: 32768
+      min: 131072
+      tcpCloseWaitTimeout: 1h0m0s
+      tcpEstablishedTimeout: 24h0m0s
+    enableProfiling: false
+    healthzBindAddress: 0.0.0.0:10256
+    hostnameOverride: ""
+    iptables:
+      masqueradeAll: false
+      masqueradeBit: 14
+      minSyncPeriod: 0s
+      syncPeriod: 30s
+    ipvs:
+      excludeCIDRs: null
+      minSyncPeriod: 0s
+      scheduler: ""
+      syncPeriod: 30s
+    kind: KubeProxyConfiguration
+    metricsBindAddress: 127.0.0.1:10249
+    mode: "iptables"
+    nodePortAddresses: null
+    oomScoreAdj: -998
+    portRange: ""
+    udpIdleTimeout: 250ms
+```
+
+```
+---
+apiVersion: apps/v1
+kind: DaemonSet
+metadata:
+  labels:
+    k8s-app: kube-proxy
+    eks.amazonaws.com/component: kube-proxy
+  name: kube-proxy
+  namespace: kube-system
+spec:
+  selector:
+    matchLabels:
+      k8s-app: kube-proxy
+  updateStrategy:
+    type: RollingUpdate
+    rollingUpdate:
+      maxUnavailable: 10%
+  template:
+    metadata:
+      labels:
+        k8s-app: kube-proxy
+    spec:
+      affinity:
+        nodeAffinity:
+          requiredDuringSchedulingIgnoredDuringExecution:
+            nodeSelectorTerms:
+            - matchExpressions:
+              - key: "kubernetes.io/os"
+                operator: In
+                values:
+                - linux
+              - key: "kubernetes.io/arch"
+                operator: In
+                values:
+                - amd64
+              # Not launching daemonset pods to fargate nodes
+              - key: "eks.amazonaws.com/compute-type"
+                operator: NotIn
+                values:
+                  - fargate
+      hostNetwork: true
+      tolerations:
+      - operator: "Exists"
+      priorityClassName: system-node-critical
+      containers:
+      - name: kube-proxy
+        image: 602401143452.dkr.ecr.REGION.amazonaws.com/eks/kube-proxy:v1.16.8
```

Comment:
Nit: Maybe we should explain that `REGION` has to be replaced. Also, do we have a list of the non-standard ECR repos that are not `602401143452`, like BAH, HKG, BJS and ZHY?

Reply:
^ @HLeeHart

Reply:
How about

```
image: ACCOUNT-ID.dkr.ecr.REGION.amazonaws.com/eks/kube-proxy:v1.16.8
# ACCOUNT-ID is the same account ID from the existing kube-proxy daemon set spec
```